src/widgets/dialogs/ReplyThreadPopup.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "messages/MessageThread.hpp"
+#include "widgets/BaseWindow.hpp"
+#include "widgets/splits/ReplyInput.hpp"
+#include "widgets/splits/Split.hpp"
+
+#include <boost/signals2.hpp>
+#include <pajlada/signals/scoped-connection.hpp>
+#include <pajlada/signals/signal.hpp>
+
+#include <memory>
+
+namespace chatterino {
+
+class ReplyThreadPopup final : public BaseWindow
+{
+    Q_OBJECT
+
+public:
+    ReplyThreadPopup(bool closeAutomatically, QWidget *parent, Split *split);
+
+    void setThread(const std::shared_ptr<const MessageThread> &thread);
+    void giveFocus(Qt::FocusReason reason);
+
+protected:
+    void focusInEvent(QFocusEvent *event) override;
+
+private:
+    void addMessagesFromThread();
+    void updateInputUI();
+
+    // The message reply thread
+    std::shared_ptr<const MessageThread> thread_;
+    // The channel that the reply thread is in
+    ChannelPtr channel_;
+    Split *split_;
```

Comment:
Can be `const` I guess.
```suggestion
    const Split *split_;
```

Reply:
`ReplyThreadPopup` uses a non-const member function of `Split` (`getChannel()`, though I'm not sure the method should be non-const).